x-pack/legacy/plugins/fleet/public/pages/agent_details/components/agent_events_table.tsx

```diff
@@ -0,0 +1,187 @@
+    setSearch,
+  };
+}
+
+function useGetAgentEvents(
+  agents: AgentsLib,
+  agent: Agent,
+  search: string,
+  page: number,
+  pageSize: number
+) {
+  const [state, setState] = useState<{ list: AgentEvent[]; total: number; isLoading: boolean }>({
+    list: [],
+    total: 0,
+    isLoading: false,
+  });
+  const fetchAgentEvents = async () => {
+    setState({
+      isLoading: true,
+      total: state.total,
+      list: state.list,
+    });
+    try {
+      const { list, total } = await agents.getAgentEvents(agent.id, search, page, pageSize);
+
+      setState({
+        isLoading: false,
+        total,
+        list,
+      });
+    } catch (err) {
+      setState({
+        isLoading: false,
+        total: 0,
+        list: [],
+      });
+    }
+  };
+  useEffect(() => {
+    fetchAgentEvents();
+  }, [agent.id, search, page, pageSize]);
+
+  return { ...state, refresh: fetchAgentEvents };
+}
+
+export const AgentEventsTable: SFC<{ agents: AgentsLib; agent: Agent }> = ({ agents, agent }) => {
+  const { pageIndex, setCurrentPageIndex, setPageSize, pageSize, pageSizes } = usePagination();
+  const { search, setSearch } = useSearch();
+
+  const { list, total, isLoading, refresh } = useGetAgentEvents(
+    agents,
+    agent,
+    search,
+    pageIndex + 1,
+    pageSize
+  );
+  const pagination = {
+    pageIndex,
+    pageSize,
+    totalItemCount: total,
+    pageSizeOptions: pageSizes,
+  };
+
+  const columns = [
+    {
+      field: 'timestamp',
+      name: i18n.translate('xpack.fleet.agentEventsList.timestampColumnTitle', {
+        defaultMessage: 'Timestamp',
+      }),
+      render: (timestamp: string) => (
+        <FormattedTime value={new Date(timestamp)} month="numeric" day="numeric" year="numeric" />
+      ),
+      sortable: true,
+    },
+    {
+      field: 'type',
+      name: i18n.translate('xpack.fleet.agentEventsList.typeColumnTitle', {
+        defaultMessage: 'Type',
+      }),
+      width: '90px',
+    },
+    {
+      field: 'subtype',
+      name: i18n.translate('xpack.fleet.agentEventsList.subtypeColumnTitle', {
+        defaultMessage: 'Subtype',
+      }),
+      width: '90px',
+    },
+    {
+      field: 'message',
+      name: i18n.translate('xpack.fleet.agentEventsList.messageColumnTitle', {
+        defaultMessage: 'Message',
+      }),
+    },
+    {
+      field: 'payload',
+      name: i18n.translate('xpack.fleet.agentEventsList.paylodColumnTitle', {
+        defaultMessage: 'Payload',
+      }),
+      truncateText: true,
+      render: (payload: any) => (
+        <span>
+          <code>{payload && JSON.stringify(payload, null, 2)}</code>
+        </span>
+      ),
+    },
+  ];
+
+  const onClickRefresh = () => {
+    refresh();
+  };
+  const onChangeQuery = (e: any) => {
+    setSearch(e.queryText);
+  };
+  const onChange = ({ page }: { page: { index: number; size: number } }) => {
+    setCurrentPageIndex(page.index);
+    setPageSize(page.size);
+  };
+
+  return (
+    <div>
```

Comment:
not a blocker but in general I prefer `<Fragment>` for this kind of thing, to avoid unnecessary DOM elements